api/src/test/java/org/openmrs/OrderTest.java

```diff
@@ -197,11 +201,22 @@ public void copy_shouldCopyAllFields() throws Exception {
 	 */
 	@Test
 	public void cloneForRevision_shouldSetAllTheRelevantFields() throws Exception {
-		assertThatAllFieldsAreCopied(new Order(), "cloneForRevision", "creator", "dateCreated", "action", "changedBy",
-		    "dateChanged", "voided", "dateVoided", "voidedBy", "voidReason", "encounter", "orderNumber", "orderer",
-		    "previousOrder", "dateActivated", "dateStopped", "accessionNumber");
+		Order newOrder = new Order();
+
+		OrderGroup orderGroup = new OrderGroup();
+		newOrder.setOrderGroup(orderGroup);
+
+		Order revisedOrder = newOrder.cloneForRevision();
+
+		assertThatAllFieldsAreCopied(revisedOrder, "cloneForRevision", "creator", "dateCreated", "action", "changedBy",
+				"dateChanged", "voided", "dateVoided", "voidedBy", "voidReason", "encounter", "orderNumber", "orderer",
+				"previousOrder", "dateActivated", "dateStopped", "accessionNumber", "orderGroup", "sortWeight");
+
+		assertNull("Revised order should not have orderGroup", revisedOrder.getOrderGroup());
```

Comment:
The expectation was that the revised order, be just a normal order, which doesn't have any ordergroup or sortweight, that is why we had not put while doing a revisionClone.

Anyways I have put it now. The setting of the value of these fields can be handled while doing a revision.